plugins/vli/fu-vli-device.c

```diff
@@ -426,6 +426,18 @@ fu_vli_device_set_kind(FuVliDevice *self, FuVliDeviceKind device_kind)
 		g_object_notify(G_OBJECT(self), "kind");
 	}
 
+	/* newer chips use SHA-256 and ECDSA-256 */
+	switch (device_kind) {
+	case FU_VLI_DEVICE_KIND_VL107:
+	case FU_VLI_DEVICE_KIND_VL650:
+	case FU_VLI_DEVICE_KIND_VL830:
+		fu_device_add_flag(FU_DEVICE(self), FWUPD_DEVICE_FLAG_SIGNED_PAYLOAD);
+		break;
+	default:
```

Comment:
rather than default, can you match older?  Otherwise a newer chip might get this flagged wrong if we forget to populate this block.  I'd rather flag an "unknown" with a warning in the daemon than misattribute it.

Reply:
There's not any device API to get this metadata yet; I've asked for some to be added. What we could do is all the other ones as another case section and then do the g_warning in the default case.

Reply:
> do the g_warning in the default case

I did that.